docs/features.md

```diff
@@ -210,6 +224,16 @@ async def response_headers():
     }
 ```
 
+
+Additionally, you can access headers for per route.
+
+```python
+@app.get("/test-headers")
+def sync_before_request(request: Request):
+    request.headers["test"] = "test-header"
+    return request
+```
```

Comment:
This will not work I think. In a route, we want to return a Response or the body of a response. If we want to change the headers of a request object, it is only relevant in a before middleware.

Reply:
> This will not work I think. In a route, we want to return a Response or the body of a response. If we want to change the headers of a request object, it is only relevant in a before middleware.


@AntoineRR , what you mean by this will not work? Yep, look at it, it is a bad example. But is working from looking at the code 😅

Reply:
Are you suggesting that we should change the example here?

Reply:
@sansyrox I just tested this and it does not crash, although the response we get from the server is something like `<builtins.Request object at 0x7f63d56f2330>` (which is the result of `str(request)`.
Also, there really is no point currently in modifying the request headers in a route, because we will not use the request in the rest of the code afterwards. Route methods take `Request` as an input and returns `Response`. The only part were it makes sense to modify request headers is in a before middleware, because the modified `Request` object will then be used inside the route method. This example would be more relevant like this IMO:

```python
@app.before_request("/test-headers") # Note this is a middleware
def test_headers_before_request(request: Request):
    request.headers["test"] = "test-header"
    return request
```

Reply:
@AntoineRR , the reason this example exists is to showcase the possibility. The ability to access headers in middlewares is shown in a section below. 

I am not concerned if it is a sensible thing to do or not, I just want to document that you can do it in Robyn.

I hope this makes sense 😅